### PR TITLE
Add MNTR aggregated failure logs to artifacts

### PR DESCRIPTION
**Pull request by IgorFedchenko**
Related to #4059

The goal of this PR is to add separate artifact containing only aggregated timeline logs for only failed specs. So that when MNTR tests are failing on CI, we could just download single small zip file and get all we need to know.

In draft state since need to debug Azure Pipelines templates update for artifacts publishing.

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@IgorFedchenko well look at that... finally looks like Azure DevOps is running properly.

**Comment by IgorFedchenko** [comment]
> @IgorFedchenko well look at that... finally looks like Azure DevOps is running properly.

@Aaronontheweb Indeed - so you did not change anything?

**Comment by IgorFedchenko** [comment]
@Aaronontheweb Also, please check out artifacts for failed MNTR here. Does it look good, or any ideas about how to make it even better?
Artifacts with failure logs are containing "FAILED" suffix.

**Comment by Aaronontheweb** [comment]
> @Aaronontheweb Indeed - so you did not change anything?

Nope, I did not - just filed a bug report and a support ticket with Azure DevOps.

**Comment by IgorFedchenko** [comment]
@Aaronontheweb Now MNTR output per each node and aggregated per spec can be found on Attachments tab in test view:
image

Do you have any ideas to make it even better, or this looks good in current state?

File: build-system/pr-validation.yaml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
       outputDirectory: 'TestResults'
       artifactName: 'net_core_tests_linux-$(Build.BuildId)'
 
-  - template: azure-pipeline.template.yaml
+  - template: azure-pipeline.mntr-template.yaml
```
Comment (IgorFedchenko) [inline build-system/pr-validation.yaml]: Have to use different template here to add build artifacts that are not used for non-mntr tests. There is some conditional execution support, but it does not seem to be flexible enough

File: build-system/pr-validation.yaml [hunk 2]
```diff
@@ -87,8 +87,10 @@ jobs:
       scriptArgs: MultiNodeTestsNetCore incremental
       outputDirectory: 'TestResults'
       artifactName: 'net_core_mntr_windows-$(Build.BuildId)'
+      mntrFailuresDir: 'TestResults\\multinode\\FAILED_SPECS_LOGS'
+      mntrFailuresArtifactName: 'net_core_mntr_FAILED_windows-$(Build.BuildId)'
```
Comment (IgorFedchenko) [inline build-system/pr-validation.yaml]: So here is where all failed spec aggregated logs can be downloaded at once

File: build-system/pr-validation.yaml (hunk 1 above)
Comment (Aaronontheweb) [reply build-system/pr-validation.yaml]: Makes sense

File: src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingGracefulShutdownSpec.cs [hunk 3]
```diff
@@ -190,6 +190,7 @@ public void ClusterSharding_should_setup_shared_journal()
             Persistence.Persistence.Instance.Apply(Sys);
             RunOn(() =>
             {
+                throw new Exception("Exception of first node");
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingGracefulShutdownSpec.cs]: Need to make sure we remove this before we merge in this PR. Was wondering why that test was failing when I was going through the Azure DevOps logs :p

File: src/core/Akka.MultiNodeTestRunner/Program.cs [hunk 4]
```diff
@@ -45,14 +46,18 @@ class Program
...
         protected static IActorRef SinkCoordinator;
 
         /// <summary>
         /// file output directory
         /// </summary>
         protected static string OutputDirectory;
+        /// <summary>
+        /// Subdirectory to store failed specs logs
+        /// </summary>
+        protected static string FailedSpecsDirectory;
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner/Program.cs]: We need to document this feature and where it's output goes here https://github.com/akkadotnet/akka.net/blob/dev/docs/articles/testing/multi-node-testing.md

Comment (Aaronontheweb) [reply src/core/Akka.MultiNodeTestRunner/Program.cs]: Maybe that can be a separate issue

**Comment by Aaronontheweb** [comment]
Since I didn't mention this anywhere: the MNTR output looks great. Much easier to be able to access the node-specific logs from the Azure DevOps UI - that should help improve contributor productivity. Nicely done @IgorFedchenko

File: src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingGracefulShutdownSpec.cs (hunk 3 above)
Comment (IgorFedchenko) [reply src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingGracefulShutdownSpec.cs]: Sure - was keeping it there to be sure that we will get some failed tests with logs

File: src/core/Akka.MultiNodeTestRunner/Program.cs (hunk 4 above)
Comment (IgorFedchenko) [reply src/core/Akka.MultiNodeTestRunner/Program.cs]: Let's put it into separate issue. Here is it: https://github.com/akkadotnet/akka.net/issues/4107

Comment (Aaronontheweb) [reply src/core/Akka.MultiNodeTestRunner/Program.cs]: Sounds good

File: src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingGracefulShutdownSpec.cs (hunk 3 above)
Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/ClusterShardingGracefulShutdownSpec.cs]: All good

**Comment by Aaronontheweb** [comment]
Hah, one small issue before I approve this - the build step fails if there are no failed MNTR specs 

```
Starting: Copy MNTR Failure Logs
==============================================================================
Task         : Copy files
Description  : Copy files from a source folder to a target folder using patterns matching file paths (not folder paths)
Version      : 2.158.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/copy-files
==============================================================================
##[error]Unhandled: Not found SourceFolder: D:\a\1\s\TestResults\multinode\FAILED_SPECS_LOGS
Finishing: Copy MNTR Failure Logs

```

**Comment by IgorFedchenko** [comment]
> Hah, one small issue before I approve this - the build step fails if there are no failed MNTR specs

Yup, seems like `CopyFiles` Azure Pipelines task can not handle case when source folder does not exist at all... Updated the pipeline.
Also, I could not find simple way to make this `FAILED_SPECS_LOGS` artifact not to be generated when no failed specs are found. But most likely it will always be generated - will see once CI will finish validation.

**Comment by IgorFedchenko** [comment]
All right, now did not failed on that step.

**Comment by Aaronontheweb** [comment]
Thanks @IgorFedchenko! Looking forward to using this in the rest of our build pipeline.

**Pull request merged**